src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala

```diff
@@ -21,8 +23,13 @@ import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchExceptionW
 
 import scala.concurrent.{ExecutionContext, Future}
 
+case class AuthorizationError(email: String) extends LeoException(s"'$email' is unauthorized", StatusCodes.Unauthorized)
+
 abstract class LeoRoutes(val leonardoService: LeonardoService, val proxyService: ProxyService, val swaggerConfig: SwaggerConfig)(implicit val system: ActorSystem, val materializer: Materializer, val executionContext: ExecutionContext) extends LazyLogging with ProxyRoutes with SwaggerRoutes with UserInfoDirectives {
 
+  private val config = ConfigFactory.parseResources("leonardo.conf").withFallback(ConfigFactory.load())
```

Comment:
It might be good to keep config stuff in the `org.broadinstitute.dsde.workbench.leonardo.config` package. You could create a WhitelistConfig case class.

Reply:
Also make the whitelist itself a Set because we’re calling contains on it.